FixedArray.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  JTFixedArray.swift
+//  
+//  A fixed length array type in Swift.
+//
+//  Created by Ben on 26/05/2016.
+
+struct JTFixedArray<T: Comparable> {
+	private(set) var length: Int = 0
+	private let defaultValue: T!
+	private var array: [T]
+	private(set) var count: Int
+	
+	init(length: Int, defaultValue: T) {
+		self.count = 0
+		self.length = length
+		self.defaultValue = defaultValue
+		self.array = [T](count: length, repeatedValue: defaultValue)
+	}
+	
+	subscript(index: Int) -> T {
+		get {
+			assert(index < length)
+			assert(index >= 0)
+			return array[index]
+		}
+		
+		set(newValue) {
+			assert(index >= 0)
+			assert(count < length)
+			count += 1
+			array[index] = newValue
+		}
+	}
+	
+	mutating func append(newElement: T) {
+		assert(count < length)
+		count += 1
+		array[count] = newElement
+	}
```

Comment:
I still have some confusion about the `append` method. For example, in the code below, the behaviour seems unusual.
```
var array = JTFixedArray(length: 4, defaultValue: 0)
array.append(1)
```
After the above code is executed, the value of the internal `array` property is `[0, 1, 0, 0]`.
Is this correct?